collectors/auth0/utils.js

```diff
@@ -0,0 +1,40 @@
+function getAPILogs(auth0Client, state, accumulator, maxPagesPerInvocation) {
+    let pageCount = 0;
+
+    let params = state.last_log_id ? { from: state.last_log_id, take:100 } : { q: "date=[" + state.since + " TO *]", per_page:100, sort: "date:1" };
+    let nextLogId = state.last_log_id ? state.last_log_id : null;
+    let lastLogTs = null;
+
+    return new Promise(function (resolve, reject) {
+        getData(params);
+        function getData(params) {
+            if (pageCount < maxPagesPerInvocation) {
+                return auth0Client
+                    .getLogs(params)
+                    .then(function (logAcc) {
+                        pageCount++;
+                        if (logAcc.length > 0) {
+                            accumulator.push(...logAcc);
+                            nextLogId = logAcc[logAcc.length - 1].log_id;
```

Comment:
I'm still wondering if we need to manually find the last log in this series. As the docs say the order of the logs is not guaranteed.

https://auth0.com/docs/logs/guides/retrieve-logs-mgmt-api#get-logs-by-checkpoint

Reply:
As per doc " The order by **date** is not guaranteed." But it's working with last_log_id and also I checked manually, logs are not duplicated. I have updated you regarding this

Reply:
Right. I remember you talking about it. I get what you're saying now. Lets leave it like this then. Thanks